lib/urn.js

```diff
@@ -50,87 +52,13 @@ module.exports = {
     }
     for (var index in urns) {
       var urn = urns[index];
-      if (!urn.match(allowedProtocolsRegExp) && urn.match(/^([^:]+:)+[^:]+$/)) {
+      var validUrnRegExp = /^([^:]+:)+[^:]+$/;
+      if (urn.match(validUrnRegExp)) {
         return true;
-      }
-
-      // Parse ID portion of URN
-      var cat = urn.split(':').slice(0, -1)
-      var id = urn.split(':').slice(-1)[0];
-
-      if (id.length % 2 !== 0) {
-        return 'Identity must have even length and be in proper format';
-      }
-
-      // Split identity by each byte
-      var charArr = id.match(/.{1,2}/g);
-
-      // Match category of URN
-      switch (cat) {
```

Comment:
I don't believe so, we enforce this at the server level (i.e. that an NFC ID or BLE ID has the right number of hex characters). Plus, in their current implementation, they were having some weird behavior and were not passing some of the unit tests (which we were not able to tell up to this point because the results were being console logged and not exiting the process). For now, it's best to just enforce that a URN has the right syntax (i.e. has ":' symbols) and let the server perform the NFC/BLE/PBK validations.